include/irr/builtin/glsl/bxdf/brdf/specular/ggx.glsl

```diff
@@ -14,21 +15,25 @@ vec3 irr_glsl_ggx_height_correlated_aniso_cos_eval(in irr_glsl_BSDFAnisotropicPa
 
     return params.isotropic.NdotL * g*ndf*fr;
 }
+*/
 //defined using NDF function with better API (compared to burley used above) and new impl of correlated smith
 vec3 irr_glsl_ggx_height_correlated_aniso_cos_eval(in irr_glsl_BSDFAnisotropicParams params, in irr_glsl_AnisotropicViewSurfaceInteraction inter, in mat2x3 ior, in float ax, in float ay)
 {
-    //float g = irr_glsl_ggx_smith_height_correlated_aniso_wo_numerator(ax, ay, params.TdotL, inter.TdotV, params.BdotL, inter.BdotV, params.isotropic.NdotL, inter.isotropic.NdotV);
     float ax2 = ax*ax;
     float ay2 = ay*ay;
-    float g = irr_glsl_ggx_smith_correlated_wo_numerator(inter.isotropic.N, params.isotropic.L, inter.isotropic.V.dir, inter.T, inter.isotropic.NdotV, params.isotropic.NdotL, ax2, ay2);
+    float g = (ax<=FLT_MIN || ay<=FLT_MIN) ? 1.0 : irr_glsl_ggx_smith_correlated_wo_numerator(
```

Comment:
do a proper if statement not a ternary operator (skipping that amount of operations will actually pay off for the GPU)

Reply:
same everywhere else

Reply:
I propose 

```glsl
float ax2=;
float ay2=;
float ndf = ;
float scalar_part = s.NdotL*ndf;
if (ax<=FLT_MIN || ay<=FLT_MIN)
{
   float g=;
   scalar_part *= g;
}
vec3 fr  = ;

return fr*scalar_part;
```

Reply:
TODO for beckmann

Reply:
do for beckmann